### PR TITLE
3.x: Update OCI integration documentation to reflect use of new OCI SDK CDI Extension

### LINKED ISSUE #4124: Update OCI integration documentation
we need to remove this from our doc "Helidon integration with Oracle Cloud Infrastructure is still experimental and not intended for production use. APIs and features have not yet been fully tested and are subject to change.".  Seems like a miss from 2.5 release perhaps.

https://helidon.io/docs/v2/#/mp/oci/01_oci

### PR DESCRIPTION
**Pull request by barchetta**
This is a forward port of #4329

These docs will likely change more, but this gets them to where they are in 2.x.

Fixes #4124

### CODE REVIEW

File: docs/mp/oci/01_oci.adoc [hunk 1]
```diff
@@ -16,63 +16,76 @@
...
-:javadoc-base-url-api: {javadoc-base-url}io.helidon.config/io/helidon/oci
+:javadoc-base-url-api: {javadoc-base-url}io.helidon.integrations.oci.sdk.cdi/io/helidon/integrations/oci/sdk/cdi/
 
 = Oracle Cloud Infrastructure Integration
 :h1Prefix: MP
 :description: Helidon OCI Integration
-:keywords: oci
-:common-deps-page-prefix-inc: ../../shared/dependencies/common_shared.adoc
+:keywords: oci, cdi
+:cdi-extension-api-url: https://jakarta.ee/specifications/cdi/2.0/cdi-spec-2.0.html#spi
```
Comment (ljnelson) [inline docs/mp/oci/01_oci.adoc:25]: If this is for `helidon-3.x` then substitute `2.0` with `3.0`

File: docs/mp/oci/01_oci.adoc [hunk 2]
```diff
@@ -16,63 +16,76 @@
...
+:jsr330-singleton-api-url: https://jakarta.ee/specifications/dependency-injection/1.0/apidocs/javax/inject/singleton
```
Comment (ljnelson) [inline docs/mp/oci/01_oci.adoc:26]: Substitute `2.0` for `1.0`

**Review by ljnelson — CHANGES_REQUESTED** [review]

**Review by romain-grecourt — APPROVED** [review]

File: docs/mp/oci/01_oci.adoc (hunk 1 above)
Comment (barchetta) [reply docs/mp/oci/01_oci.adoc]: fixed

File: docs/mp/oci/01_oci.adoc (hunk 2 above)
Comment (barchetta) [reply docs/mp/oci/01_oci.adoc]: fixed

**Review by ljnelson — APPROVED** [review]

**Pull request merged**